frontend/ui/widget/radiomark.lua

```diff
@@ -0,0 +1,34 @@
+local BD = require("ui/bidi")
+local Blitbuffer = require("ffi/blitbuffer")
+local Font = require("ui/font")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local TextWidget = require("ui/widget/textwidget")
+
+local RadioMark = InputContainer:new{
+    checkable = true, -- empty space when false
+    checked = false,
+    enabled = true,
+    face = Font:getFace("smallinfofont"),
+    baseline = 0,
+    _mirroredUI = BD.mirroredUILayout(),
+}
+
+local BUTTON = {
+    "\u{25C9} ", -- FISHEYE (big dot)
+    "\u{29BF} ", -- CIRCLED BULLET (small dot)
+    "\u{25EF} ", -- LARGE CIRCLE (empty)
+}
+
+function RadioMark:init()
+    local widget = TextWidget:new{
+        text = self.checkable and (self.checked and BUTTON[2] or BUTTON[3]) or "",
```

Comment:
May be once you have ecided on the good symbols, hardcode them here, or use BUTTON = { CHECKED=one, UNCHECKED=theother } so you don't need to use array indexes here (BUTTON[2]), for better readability.